src/custom/utils/gas.ts

```diff
@@ -0,0 +1,105 @@
+import { ChainId } from '@uniswap/sdk'
+import { BigNumber } from 'ethers'
+import { UpdateGasPrices } from 'state/gas/actions'
+import { DEFAULT_NETWORK_FOR_LISTS } from 'constants/lists'
+import { fetchData, promiseFirstResolved } from '.'
+import { getChainIdValues } from './misc'
+import { parseUnits } from 'ethers/lib/utils'
+
+export enum GasPriceSources {
+  SAFE = 'SAFE',
+  INFURA = 'INFURA'
+}
+
+export interface GasPriceEndpointData {
+  source: GasPriceSources
+  options?: RequestInit
+  urls: { [chain in ChainId]: string }
+}
+
+export function chainIdToGasLabel(chainId: ChainId) {
+  switch (chainId) {
+    case ChainId.RINKEBY:
+      return 'rinkeby'
+    case ChainId.GÖRLI:
+      return 'goerli'
+    default:
+      return 'mainnet'
+  }
+}
+
+// safe relay endpoint
+export const GAS_FEE_ENDPOINTS: GasPriceEndpointData = {
+  source: GasPriceSources.SAFE,
+  urls: getChainIdValues().reduce((memo, chainId) => {
```

Comment:
I personally feel is a bit overkill these reduce. I would feek is easier to have the baseUrl per network in a plain simple object instead of making it more obscure inside a reduce logic. It's not like we add networks every day, and makes it harder just by looking at this which network every gasPrice provider support. 


one plain object will just return undefined if that network don't have baseUrl, then u know u cannot use it.